### PR TITLE
Create / Get Plan use cases

### PR DESCRIPTION
**Pull request by kosiakkatrina**
This PR contains the initial Get / Create use cases for creating and getting a shared plan. 

Contents:

- CreatePlan use-case
- GetPlan use-case
- Plan Gateway
- Plan domain object
- Create/Get api routes
- Unit tests for all of the above 

The db related code is based on spikes done against a dynamo db local instance.

### CODE REVIEW

File: lib/dependencies.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import GetPlan from './use-cases/getPlan';
+import CreatePlan from './use-cases/createPlan';
+import PlanGateway from './gateways/planGateway';
+import AWS from 'aws-sdk';
+
+const config = {
+  region: 'localhost',
+  endpoint: 'http://localhost:8000',
+  accessKeyId: 'foo',
+  secretAccessKey: 'bar'
```
Comment (t-davies) [inline lib/dependencies.js:10]: Do we want these as environment variables, or conditional based on the environment maybe?

File: package.json [hunk 2]
```diff
@@ -5,11 +5,29 @@
   "scripts": {
     "dev": "next dev",
     "build": "next build",
-    "start": "next start"
+    "start": "next start",
+    "unit-test": "jest --watch",
+    "lint": "./node_modules/eslint/bin/eslint.js --fix --ext js lib/ pages/ && echo 'Lint complete.'"
```
Comment (t-davies) [inline package.json:10]: Just `eslint` on it's own should also work?
```
eslint --fix --ext js lib/ pages/ && echo 'Lint complete.'
```

File: pages/api/createPlanApi.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+        firstName: req.body.firstName,
+        lastName: req.body.lastName
+      });
+
+      res.status(200).json(result);
+    } catch (err) {
+      //log error here
+      res.status(500).json({
+        error: `could not create a plan with first name: ${req.body.firstName}, last name: ${req.body.lastName}`
+      });
```
Comment (t-davies) [inline pages/api/createPlanApi.js:16]: Maybe worth differentiating between validation problems (e.g. we throw an error if `lastName` is null) and then returning a 400 instead? Could be useful on the front-end to determine what sort of error message to show?

Comment (mmmmillar) [reply pages/api/createPlanApi.js:16]: yeah that sounds like a good idea

File: lib/dependencies.js (hunk 1 above)
Comment (mmmmillar) [reply lib/dependencies.js:10]: ah yep i missed this, will take a look

**Pull request merged**